src/middleware/builtin.ts

```diff
@@ -206,18 +206,29 @@ export function matchConstraints(
   };
 }
 
+const messagePostedEventSubtypesAsArray = [undefined, 'bot_message', 'file_share', 'thread_broadcast'];
+
 /*
  * Middleware that filters out messages that don't match pattern
  */
-export function matchMessage(
+export function matchMessage<
+  Subtypes extends string | undefined = string | undefined,
+>(
   pattern: string | RegExp,
-): Middleware<SlackEventMiddlewareArgs<'message' | 'app_mention'>> {
```

Comment:
having app_mention here had been false as a preceding built-in middleware `matchEventType('message')` does not accept app_mention. Moreover, it should not accept two types of events. So, I've removed it.